bfrec

```diff
@@ -146,6 +146,53 @@ fi
 bfb_location=/lib/firmware/mellanox/boot
 capsule_location=/lib/firmware/mellanox/boot/capsule
 
+parse_and_set_capsule_versions()
+{
+    local capsule_file="$1"
+
+    if [ -z "$capsule_file" ]; then
+        echo "ERROR: No capsule file specified"
+        exit 2
+    fi
+
+    if [ ! -f "$capsule_file" ]; then
+        echo "ERROR: Capsule file '$capsule_file' not found"
+        exit 2
+    fi
+
+    # Use bfver to parse the ATF and UEFI version from the capsule file.
+    BFVER_OUTPUT=$(bfver -f "$capsule_file")
+
+    local atf_version=""
+    local uefi_version=""
+    while IFS= read -r line; do
+        if [[ $line == *"BlueField ATF version:"* ]]; then
+            atf_version=$(echo "$line" | cut -d':' -f3 | xargs)
+        elif [[ $line == *"BlueField UEFI version:"* ]]; then
+            uefi_version=$(echo "$line" | cut -d':' -f2 | xargs)
+        fi
+    done <<< "$BFVER_OUTPUT"
+
+    # Check if both versions were successfully parsed
+    if [ -n "$atf_version" ] && [ -n "$uefi_version" ]; then
+        echo "Parsed versions: ATF=$atf_version, UEFI=$uefi_version"
+
+        # Transfer the ATF and UEFI pending version to UEFI.
+        bfcfg --capatfver "$atf_version"
+        bfcfg --capuefiver "$uefi_version"
```

Comment:
Question: does `bfcfg` command logs the processing of `--capatfver` and `--capuefiver` ?

From developer/tester perspective, how do I know if the versions are correctly passed to UEFI/ATF (meaning the special variable is written)?

I would have thought to print a message if these operations passed. So perhaps, we relocate the echo at line 178 to line 183?

Reply:
It looks like we have double `echo` one at line 184 and a second at line 196. It is intentional?

If no, I suggest we relocate the `echo` from line 184 to line 196; as I believe `Boot image detected: ATF pending version: $atf_version, UEFI pending version: $uefi_version` is more generic and simpler than `Boot image updated to UEFI...`. I think "updated to UEFI" is an implementation detail and it is meaningless to the user. If tomorrow the interface changes, the message would change. Plus, the boot image isn't updated yet :)